### ISSUE TITLE
checking for model feasibility following solve

### ISSUE THREAD

**Issue by jGaboardi** [pysal/spopt#238]
Following up from #237, we should probably be checking to the solution status following a `.solve()` attempt. This is because `pulp` does not raise an exception when model is not solved. However, a [status code](https://coin-or.github.io/pulp/technical/constants.html#pulp.constants.LpStatus) can be retrieved through the `status` attribute following a solution attempt:

      
|  LpStatus  key              | string value  | numerical value |
|-----------------------|-------------|-----------------|
|  `LpStatusOptimal`    | "Optimal"     |          1      |
| `LpStatusNotSolved`  | "Not Solved"  |          0      |
|  `LpStatusInfeasible` | "Infeasible"  |         -1      |
|  `LpStatusUnbounded`  | "Unbounded"   |          -2     |
|  `LpStatusUndefined`  | "Undefined"   |          -3     |

Basically, we'll want to check the status and raise an exception (or at the very least a warning) if the status is anything but `1`. See here for a [gist demonstrating an example](https://gist.github.com/jGaboardi/9128fc7cda3e04d1840fd9c64c24e759).

cc @qszhao
